woocommerce/woocommerce-android
# Refactor VariationDetailRepository into stateless repo

Closes: #6064 


Remove the "do not merge" label after merging [the FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2325) and updating the hashcode in build.gradle.

This PR refactors fetchSingleVariation into a suspendable function - removes usage of the event bus.

There are two reasons for this change

1. [The event bus architecture is considered as legacy/deprecated](https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/%5BDeprecated%5D-Architecture)
2. WCAndroid app uses stateful repositories which need to be registered to the event bus - this leads to memory leaks when the repository doesn't unregister from the event bus. It also causes side-effects when multiple instances of a repository are created - eg. duplicate tracking events.


## How to test
1. Open Variation Product
2. Open List of variations
3. Go to web -> change something on one of the variations - eg. price
4. In the app, open detail of this variation -> notice it shows the original price for a bit but it updates it automatically
